packages/contracts/contracts/SemaphoreVoting.sol

```diff
@@ -0,0 +1,103 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.23;
+
+import {ISemaphore} from "@semaphore-protocol/contracts/interfaces/ISemaphore.sol";
+import {ISemaphoreVoting} from "./interfaces/ISemaphoreVoting.sol";
+import {SemaphoreGroups} from "@semaphore-protocol/contracts/base/SemaphoreGroups.sol";
+
+/// @title Semaphore Voting contract.
+/// @notice It allows users to vote anonymously in a poll.
+/// @dev The following code allows you to create polls, add voters and allow them to vote anonymously.
+contract SemaphoreVoting is ISemaphoreVoting, SemaphoreGroups {
+    ISemaphore public group;
+
+    /// @dev Gets a poll id and returns the poll data.
+    mapping(uint256 => Poll) internal polls;
+
+    /// @dev Checks if the poll coordinator is the transaction sender.
+    /// @param pollId: Id of the poll.
+    modifier onlyCoordinator(uint256 pollId) {
+        if (polls[pollId].coordinator != msg.sender) {
+            revert Semaphore__CallerIsNotThePollCoordinator();
+        }
+
+        _;
+    }
+
+    /// @dev Initializes the Semaphore group used to verify the user's ZK proofs.
+    /// @param _group: Semaphore group address.
+    constructor(ISemaphore _group) {
+        group = _group;
+    }
+
+    /// @dev See {ISemaphoreVoting-createPoll}.
+    function createPoll(uint256 pollId, address coordinator) public override {
```

Comment:
We can get rid of the `pollId` parameter here. We can use the `groupId` returned by `createGroup()` directly.